File: src/main/java/io/github/thehrz/snowcraft/object/misc/IceCream.java
package io.github.thehrz.snowcraft.object.misc;

import io.github.thehrz.snowcraft.list.Items;
import io.izzel.taboolib.module.inject.TListener;
import me.mrCookieSlime.Slimefun.Setup.SlimefunManager;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerInteractEvent;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;

import static io.izzel.taboolib.module.locale.TLocale.Display.sendActionBar;

@TListener
public class IceCream implements Listener {
    @EventHandler
    public void onPlayerInteract(PlayerInteractEvent event) {
        if (SlimefunManager.isItemSimiliar(event.getItem(), Items.ICE_CREAM, true)) {
            event.setCancelled(true);
            int amount = event.getPlayer().getInventory().getItemInMainHand().getAmount();
            event.getPlayer().getInventory().getItemInMainHand().setAmount(--amount);
            sendActionBar(event.getPlayer(), "§a你食用了冰淇淋 获得了冰淇淋给予你的跳跃加成");
            event.getPlayer().addPotionEffect(new PotionEffect(PotionEffectType.JUMP, 3600, 5));
        }
    }
}

File: src/main/kotlin/io/github/thehrz/snowcraft/object/misc/IceCream.kt
package io.github.thehrz.snowcraft.`object`.misc

import io.github.thehrz.snowcraft.list.Items
import me.mrCookieSlime.Slimefun.Setup.SlimefunManager
import org.bukkit.event.player.PlayerInteractEvent
import org.bukkit.potion.PotionEffect
import org.bukkit.potion.PotionEffectType
import taboolib.common.platform.event.SubscribeEvent
import taboolib.platform.util.sendLang

object IceCream {
    @SubscribeEvent
    fun e(e: PlayerInteractEvent) {
        if (SlimefunManager.isItemSimiliar(e.item, Items.ICE_CREAM, true)) {
            e.let {
                it.isCancelled = true
                it.item.amount -= 1
                it.player.sendLang("ice-cream-eating")
                it.player.addPotionEffect(PotionEffect(PotionEffectType.JUMP, 3600, 5))
            }
        }
    }
}